src/js/hooks/list-pages/productSupplier/useProductSupplierActions.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useCallback } from 'react';
+
+import { confirmAlert } from 'react-confirm-alert';
+import { RiDeleteBinLine, RiPencilLine } from 'react-icons/ri';
+import { useSelector } from 'react-redux';
+
+import productSupplierApi from 'api/services/ProductSupplierApi';
+import { PRODUCT_SUPPLIER_URL } from 'consts/applicationUrls';
+import CustomConfirmModal from 'utils/CustomConfirmModal';
+import { hasPermissionsToProductSourceActions } from 'utils/permissionUtils';
+
+import 'react-confirm-alert/src/react-confirm-alert.css';
+
+const useProductSupplierActions = ({ fireFetchData }) => {
+  const { currentUser, isAdmin } = useSelector((state) => ({
+    currentUser: state.session.user,
+    isAdmin: state.session.isUserAdmin,
+  }));
+
+  const deleteProductSupplier = async (onClose, productSupplierId) => {
+    try {
+      await productSupplierApi.deleteProductSupplier(productSupplierId);
+    } finally {
+      fireFetchData?.();
```

Comment:
I'm not sure if we want to refetch the data if something goes wrong with the delete. I think it should be placed inside the `try`. But I do not have a strong opinion on that.
Since we return `204` no matter what, the only exception I might imagine would be if we lost the connection.